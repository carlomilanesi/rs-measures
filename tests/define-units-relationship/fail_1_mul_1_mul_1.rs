rs_measures::define_measure_types! {
    MeasureFeatures {
        with_points: false,
        with_directions: false,
        with_2d: false,
        with_3d: false,
        with_transformations: false,
        with_uncertainty: None,
    }
}

pub struct P1;
impl VectorProperty for P1 {}

pub struct U1;
impl MeasurementUnit for U1 {
    type Property = P1;
    const RATIO: f64 = 1.;
    const OFFSET: f64 = 0.;
    const SUFFIX: &'static str = " u1";
}

pub struct P2;
impl VectorProperty for P2 {}

pub struct U2;
impl MeasurementUnit for U2 {
    type Property = P2;
    const RATIO: f64 = 1.;
    const OFFSET: f64 = 0.;
    const SUFFIX: &'static str = " u2";
}

pub struct P3;
impl VectorProperty for P3 {}

pub struct U3;
impl MeasurementUnit for U3 {
    type Property = P3;
    const RATIO: f64 = 1.;
    const OFFSET: f64 = 0.;
    const SUFFIX: &'static str = " u3";
}

rs_measures::define_units_relationship! { U1 * U2 * U3 }

fn main() {}
