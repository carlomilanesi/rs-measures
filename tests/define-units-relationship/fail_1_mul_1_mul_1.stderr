error: no rules expected the token `*`
  --> tests/define-units-relationship/fail_1_mul_1_mul_1.rs:45:46
   |
45 | rs_measures::define_units_relationship! { U1 * U2 * U3 }
   |                                              ^ no rules expected this token in macro call
   |
note: while trying to match `==`
  --> src/define_units_relationship.rs
   |
   |     { $id1:ident == $id2:ident * $id3:ident} => { rs_measures::expand_1_1! {$id2 $id3 $id1} };
   |                  ^^
